tests/ValueObjectTest.php

```diff
@@ -125,6 +126,35 @@ public function classes_are_supported()
         };
     }
 
+    /** @test */
+    public function generic_collections_are_supported()
+    {
+        new class(['foo' => [new DummyClass(), new DummyClass(), new DummyClass()]]) extends ValueObject {
+            /** @var \Spatie\ValueObject\Tests\TestClasses\DummyClass[] */
+            public $foo;
+        };
+
+        $this->markTestSucceeded();
+
+        $this->expectException(ValueObjectException::class);
+
+        new class(['foo' => [new DummyClass(), new DummyClass(), new OtherClass()]]) extends ValueObject {
+            /** @var \Spatie\ValueObject\Tests\TestClasses\DummyClass[] */
+            public $foo;
+        };
+    }
+
+    /** @test */
+    public function an_exception_is_thrown_for_a_generic_collection_of_null()
+    {
+        $this->expectException(ValueObjectException::class);
+
+        new class(['foo' => [null]]) extends ValueObject {
+            /** @var null[] */
```

Comment:
This isn't what I meant with "null should not be supported", I rather meant this:

```php
new class(['foo' => [null]]) extends ValueObject {
    /** @var string[] */
    public $foo;
};
```

Meaning that typed arrays may not contain `null` values.

Reply:
Ah, was thinking to literal I guess :p